experimental/dds/tree2/src/feature-libraries/chunked-forest/codec/format.ts

```diff
@@ -14,6 +16,32 @@ import {
 
 export const version = "unstable-development";
 
+// Compatible versions used for format/version validation.
+// If a new version is created, be sure to remove incompatible versions from this set.
+const validVersions = new Set([version]);
+
+const Versioned = Type.Object({
+	version: Type.String(),
+});
+type Versioned = Static<typeof Versioned>;
+
+export const versionValidator = typeboxValidator.compile(Versioned);
+
+/**
+ * Validates the version and format of encoded data against a given schema.
+ *
+ * @param encodedData - The data to validate.
+ * @param schema - The TypeBox schema to validate against.
+ */
+export function validateFormat<TEncoded>(encodedData: TEncoded, schema: TSchema) {
+	assert(versionValidator.check(encodedData), "Encoded data must be versioned.");
+	assert(validVersions.has(encodedData.version), "Encoded data version is unknown.");
+
+	// Compile the format schema and validate the data against it.
+	const formatValidator = typeboxValidator.compile(schema);
```

Comment:
`typeboxValidator` should not be statically referenced by tree2: it adds a reasonable amount of bundle size which customers may not want to always pay.

Instead you should follow patterns similar to other codecs--take in a `JsonValidator` implementation and use that. Ultimately this will come from the options used to initialize the `SharedTree`.

Reply:
additionally, compiling a schema into a validation function is generally a nontrivial operation. You should ideally compile it a single time, not each time data is validated.

Reply:
replaced with codecs which takes in `JsonValidator`.